section1/set8/exercises/07-multiplication-table-exercises/02-math-tables/main.go

```diff
@@ -105,4 +113,57 @@ package main
 // ---------------------------------------------------------
 
 func main() {
+	if len(os.Args) != 3 {
+		fmt.Println("Usage: [op = %*/+-] [size]")
+		return
+	}
+
+	os1 := strings.IndexAny(os.Args[1], "%*-/+")
+
+	if os1 == -1 {
+		fmt.Println("Invalid operator.\nValid ops are: %+-*/")
+		return
+	}
+
+	n, err := strconv.Atoi(os.Args[2])
+
+	if err != nil {
+		fmt.Println("Incorrect size value.\nUsage: [op = %*/+-] [size]")
+		return
+	}
+
+	fmt.Printf("%5s", os.Args[1])
+	for i := 0; i <= n; i++ {
+		fmt.Printf("%5d", i)
+	}
+	fmt.Println()
+
+	for i := 0.; i <= float64(n); i++ {
+		fmt.Printf("%5.0f", i)
+		for j := 0.; j <= float64(n); j++ {
+			m := math.Mod(i, j)
+			switch os.Args[1] {
```

Comment:
always copy the os.args into local arr before use; like args := os.args[1:]
This simplifies the logic

Reply:
fixed